File: src/GlassCS/Glass/Actions/ToggleVisibilityAction.cs
using System.Windows;
using System.Windows.Interactivity;

namespace ESRI.ArcGIS.SilverlightMapApp.Actions
{
    public class ToggleVisibilityAction : TargetedTriggerAction<UIElement>
    {
        protected override void Invoke(object parameter)
        {
            this.Target.Visibility = this.Target.Visibility == Visibility.Visible ?
                Visibility.Collapsed : Visibility.Visible;
        }
    }
}

File: src/GlassVB/GlassVB/Actions/ToggleVisibilityAction.vb
Imports System.Windows
Imports System.Windows.Interactivity

Namespace ESRI.ArcGIS.SilverlightMapApp.Actions
		Public Class ToggleVisibilityAction
			Inherits TargetedTriggerAction(Of UIElement)
				Protected Overrides Sub Invoke(ByVal parameter As Object)
						Me.Target.Visibility = If(Me.Target.Visibility = Visibility.Visible, Visibility.Collapsed, Visibility.Visible)
				End Sub
		End Class
End Namespace